server/lib/src/idm/application.rs

```diff
@@ -240,6 +244,68 @@ impl GenerateApplicationPasswordEvent {
     }
 }
 
+impl IdmServerProxyWriteTransaction<'_> {
+    #[instrument(level = "debug", skip_all)]
+    pub fn generate_application_password(
+        &mut self,
+        ev: &GenerateApplicationPasswordEvent,
+    ) -> Result<(String, Uuid), OperationError> {
+        // This is intended to be read/copied by a human
+        let cleartext = readable_password_from_random();
+        let policy = self.crypto_policy();
+
+        let ap = ApplicationPassword::new(
+            ev.application,
+            ev.label.as_str(),
+            cleartext.as_str(),
+            policy,
+        )
+        .inspect_err(|err| error!(?err, "Unable to generate application password"))?;
```

Comment:
Is there a why/actionable outcome to this for the admin? I'm guessing "file a bug" but it should be clearer and/or give a reason.